diofant/integrals/tests/test_meijerint.py

```diff
@@ -651,3 +651,8 @@ def test_issue_6860():
 def test_issue_8368():
     assert meijerint_indefinite(cosh(x)*exp(-x*t), x) == (
         (-t - 1)*exp(x) + (-t + 1)*exp(-x))*exp(-t*x)/2/(t**2 - 1)
+
+
+def test_issue_4311():
+    assert meijerint_indefinite(x*abs(9 - x**2), x) == \
+        -81*meijerg(((3, 1), (2,)), ((1,), (2, 0)), x**2/9)
```

Comment:
> Rewriting the coefficient only made it possible for SymPy to continue the computation without exceeding the maximum recursion depth.

An exception is bad, but you miss that is not a wrong answer.  In turn, with this "continuation of the computation" now you got a number of wrong answers instead of such exceptions (or nan's).

Just a few new "answers" you introduce:
https://github.com/sympy/sympy/issues/8733
https://github.com/sympy/sympy/issues/7165
https://github.com/sympy/sympy/issues/4311

New code should be covered with tests, isn't?

> It would be interesting to search for the bug

Well, there may be more then one.
1) meijerint_indefinite returns the wrong answer (unless mpmath did numerical computations wrongly in this case).
2) hyperexpand can't simplify answer (e.g. to Mathematica's one).
3) Mathematica can be incorrect, evaluating meijerg from 1). btw, it returns ``81 MeijerG[{{3}, {}}, {{}, {0}}, x^2/9]`` on the WolframAlpha too.